tests/bsim/bluetooth/host/scan/start_stop/src/main.c

```diff
@@ -0,0 +1,266 @@
+#include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/bluetooth/addr.h>
+#include <zephyr/sys/atomic.h>
+#include <zephyr/sys/atomic_builtin.h>
+#include <babblekit/testcase.h>
+#include <babblekit/sync.h>
+
+LOG_MODULE_REGISTER(bt_bsim_scan_start_stop, LOG_LEVEL_DBG);
+
+#define WAIT_TIME_S 60
+#define WAIT_TIME   (WAIT_TIME_S * 1e6)
+
+static atomic_t flag_adv_report_received;
+static atomic_t flag_periodic_sync_established;
+static bt_addr_le_t adv_addr;
+
+static void test_tick(bs_time_t HW_device_time)
+{
+	if (bst_result != Passed) {
+		TEST_FAIL("Test failed (not passed after %d seconds)\n", WAIT_TIME_S);
+	}
+}
+
+static void test_init(void)
+{
+	bst_ticker_set_next_tick_absolute(WAIT_TIME);
+}
+
+static void bt_sync_established_cb(struct bt_le_per_adv_sync *sync,
+			    struct bt_le_per_adv_sync_synced_info *info)
+{
+	LOG_DBG("Periodic sync established");
+	atomic_set(&flag_periodic_sync_established, true);
+}
+
+static struct bt_le_per_adv_sync_cb sync_callbacks = {
+	.synced = bt_sync_established_cb,
+};
+
+static void device_found(const bt_addr_le_t *addr, int8_t rssi, uint8_t type,
+			 struct net_buf_simple *ad)
+{
+	char addr_str[BT_ADDR_LE_STR_LEN];
+
+	memcpy(&adv_addr, addr, sizeof(adv_addr));
+
+	bt_addr_le_to_str(&adv_addr, addr_str, sizeof(addr_str));
+	LOG_DBG("Device found: %s (RSSI %d), type %u, AD data len %u",
+	       addr_str, rssi, type, ad->len);
+	atomic_set(&flag_adv_report_received, true);
+}
+
+void run_dut(void)
+{
+	/* Test purpose:
+	 *
+	 * Verifies that the scanner can establish a sync to a device when
+	 * it is explicitly enabled and disabled. Disabling the scanner
+	 * explicitly should not stop the implicitly started scanner.
+	 * Verify that the explicit scanner can be started when the
+	 * implicit scanner is already running.
+	 *
+	 * Two devices:
+	 * - `dut`: tries to establish the sync
+	 * - `peer`: runs an advertiser / periodic advertiser
+	 *
+	 * Procedure:
+	 * - [dut] start establishing a sync (no peer)
+	 * - [peer] starts advertising
+	 * - [dut] starts explicit scanning
+	 * - [dut] wait for the peer to be found
+	 * - [dut] stops explicit scanning
+	 * - [dut] stop the periodic sync
+	 * - [dut] start establishing a sync to the peer
+	 * - [dut] start and stop explicit scanning
+	 * - [peer] start periodic advertiser
+	 * - [dut] wait until a sync is established
+	 *
+	 * [verdict]
+	 * - dut is able to sync to the peer.
+	 */
+
+	LOG_DBG("start");
+	bk_sync_init();
+	int err;
+
+	LOG_DBG("Starting DUT");
+
+	err = bt_enable(NULL);
+	TEST_ASSERT(!err, "Bluetooth init failed (err %d)\n", err);
+
+	LOG_DBG("Bluetooth initialised");
+
+	/* Try to establish a sync to a periodic advertiser.
+	 * This will start the scanner.
+	 */
+	memset(&adv_addr, 0x00, sizeof(adv_addr));
+	struct bt_le_per_adv_sync_param per_sync_param = {.addr = adv_addr,
+							  .options = 0x0,
+							  .sid = 0x0,
+							  .skip = 0x0,
+							  .timeout = BT_GAP_PER_ADV_MAX_TIMEOUT};
```

Comment:
If you are using something like `clang-format` then if you end the final field with `,` then it formats it in a way that at least I think looks better, but that's just a personal preference :)

Reply:
nice trick :+1:

Reply:
Just FYI, the comment also applied to the other similar places :)